Sources/ResilientDecoding/Resilient.swift

```diff
@@ -35,6 +35,7 @@ public struct Resilient<Value: Decodable>: Decodable {
     topLevelError: Error?,
     errorsAtOffset: [ErrorAtOffset])
   {
+    assert(topLevelError != nil || !errorsAtOffset.isEmpty)
```

Comment:
Is this a breaking change? Or would this have asserted elsewhere before?

Reply:
This is a bit strange. Shouldn't an API that takes an optional be ok when an optional is passed?

Reply:
I agree with the comment from Dan below, this def sounds like an enum in the making 😉